File: 268.missing-number.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        dictionary = {}
        for i in range(len(nums) + 1):
            dictionary[i] = True

        for num in nums:
            dictionary.pop(num)

        for key in dictionary:
            return key
        

        

File: 268-missing-number/268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        long long n = nums.size();
        long long sum = (n+1) * n / 2;
        long long curr_sum = 0;
        for (auto num: nums) {
            curr_sum += num;
        }
        return sum - curr_sum;
        
    }
};